TestStateTransitionAppFailureHandling is not stable

### Describe the bug
The mentioned test is not stable. And it fails occasionally in the Github test job.

2020-11-24T02:00:58.2739569Z Shut down zookeeper at port 2183 in thread main
2020-11-24T02:01:01.1044299Z [ERROR] Tests run: 1250, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 5,186.041 s <<< FAILURE! - in TestSuite
2020-11-24T02:01:01.1058244Z [ERROR] testSTHandlerInitFailureRetry(org.apache.helix.integration.paticipant.TestStateTransitionAppFailureHandling)  Time elapsed: 1.054 s  <<< FAILURE!
2020-11-24T02:01:01.1067145Z java.lang.NullPointerException
2020-11-24T02:01:01.1081750Z 	at org.apache.helix.integration.paticipant.TestStateTransitionAppFailureHandling.lambda$testSTHandlerInitFailureRetry$1(TestStateTransitionAppFailureHandling.java:105)
2020-11-24T02:01:01.1098732Z 	at org.apache.helix.integration.paticipant.TestStateTransitionAppFailureHandling.testSTHandlerInitFailureRetry(TestStateTransitionAppFailureHandling.java:96)
2020-11-24T02:01:01.1103523Z 
2020-11-24T02:01:01.5205487Z [ERROR] Failures: 
2020-11-24T02:01:01.5208508Z [ERROR]   TestStateTransitionAppFailureHandling.testSTHandlerInitFailureRetry:96->lambda$testSTHandlerInitFailureRetry$1:105 NullPointer
2020-11-24T02:01:01.5210731Z [ERROR] Tests run: 1250, Failures: 1, Errors: 0, Skipped: 1

### To Reproduce
For now, it is only observed in the Github test job.

### Expected behavior
Pass stably.

### Additional context
Add any other context about the problem here.

--- a/helix-core/src/test/java/org/apache/helix/integration/paticipant/TestStateTransitionAppFailureHandling.java
+++ b/helix-core/src/test/java/org/apache/helix/integration/paticipant/TestStateTransitionAppFailureHandling.java
@@ -102,7 +102,7 @@ public class TestStateTransitionAppFailureHandling extends ZkStandAloneCMTestBas
                 .map(childName -> keyBuilder.message(instanceName, childName))
                 .collect(Collectors.toList()), true);
         for (Message message : messageList) {
-          if (message.getMsgState() != Message.MessageState.UNPROCESSABLE) {
+          if (message == null || message.getMsgState() != Message.MessageState.UNPROCESSABLE) {
             return false;
           }
         }

